### ISSUE TITLE
Nteract not working since update to ipykernel

### ISSUE THREAD

**Issue by joecorkerton** [nteract/archived-desktop-app#1640]
Yesterday I updated my Python packages, which included updating ipykernel to 4.6.0. Since then I have been unable to execute any cells in nteract, they just run endlessly. What is strange is that nteract does seem to find the kernel as it is listed under the options menu and it does not pop up an error message that it cannot be found. It also still works fine with my Julia kernel so problem seems to be solely related to Python 3.

I have tried reinstalling all python packages via anaconda and also reinstalling the kernel, which is correctly listed in `jupyter kernelspec list --json`

**Comment by bgarchow** [comment]
I have just installed nteract for Mac and I also cannot execute any code. I have python3 kernel (4.6.1) and R kernel (0.8.6.9) installed and running fine in the browser notebook. nteract lists both kernels under the Languages menu but will not execute code for either language. I also do not have syntax coloring for either language.

**Comment by rgbkrk** [comment]
This is pretty strange to me, I upgraded my ipykernel and it's working fine. Are you using Anaconda as well @bgarchow?

**Comment by bgarchow** [comment]
@rgbkrk no I am not using anaconda. I have python3 installed from homebrew and R install is from the CRAN binary package.

**Comment by joecorkerton** [comment]
@rgbkrk any idea of how I can get nteract working again? Just tried a fresh install of nteract, with no effect

**Comment by rgbkrk** [comment]
I think we're going to have to get what we can from the dev tools. I'd be happy to screenshare with you to try to figure it out. My other suggestion would be to re-run the `python -m ipykernel install --user` step and fully restart nteract.

**Comment by rgbkrk** [comment]
We just did a screenshare and made a new kernelspec which only differed in name (folder name) and display name (in the `kernel.json`). I'm at a loss for words for why the kernel was hanging. The only thing I can see different in this setup is that `python` was the root conda environment.

**Comment by joecorkerton** [comment]
@rgbkrk I have managed to resolve the issue! It seems there was another kernelspec in `/anaconda3/share/jupyter/kernels/` which wasn't working correctly, and produced a 'false positive' for both Nteract and Jupyter notebook. I only discovered this kernelspec after deleting the one in `/.local/share/jupyter/kernels/`, and I am not sure how this secondary kernelspec came about.

It seems Jupyter and Nteract had slightly different behaviour in this instance as Jupyter defaulted to the correct kernelspec before it was deleted but Nteract didn't
